### PR TITLE
feat(QTM-312): Novo componente ChargeBar

### PR DESCRIPTION
**Pull request by rapahaeru**
## Description
https://jirasoftware.catho.com.br/browse/QTM-312

## Setup
to view the components behavior, run `yarn storybook`

## Review guide
- [ ] Coverage (coverage status should not regress)\
      - `yarn test:coverage`
- [ ] Unit tests (yarn test:components)
- [ ] Regression \
      - first start the storybook for regression tests(and keep it open): ` yarn test:regression:storybook`; \
      - then run the regression tests: `yarn test:regression`
- [ ] Code review
- [ ] TypeScript updated
- [ ] a11y

### Browsers review
- [ ] Layout review
  - [ ] Edge
  - [ ] Firefox
  - [ ] Chrome
  - [ ] Safari
  - [ ] Mobile
- [ ] Ux review validation

### CODE REVIEW

File: components/ChargeBar/ChargeBar.jsx [hunk 1]
```diff
@@ -0,0 +1,151 @@
...
+ChargeBar.propTypes = {
+  skin: PropTypes.oneOf(['neutral', 'primary', 'secondary', 'success']),
+  theme: PropTypes.shape({
+    components: PropTypes.shape({
+      chargeBar: PropTypes.object,
+    }),
+    spacing: PropTypes.object,
+    baseFontSize: PropTypes.number,
+  }),
+  width: PropTypes.string,
```
Comment (toncabral88) [inline components/ChargeBar/ChargeBar.jsx:146]: What do you think about change this proptype to number? In this way, I believe that implementer can send other units than px or send only number without px.

File: components/ChargeBar/ChargeBar.jsx [hunk 2]
```diff
@@ -0,0 +1,151 @@
...
+
+const Wrapper = styled.div`
+  display: flex;
+  flex-direction: column;
+  font-stretch: normal;
+  font-style: normal;
+  font-weight: normal;
+  letter-spacing: 0.18px;
+  line-height: 1.5;
+`;
```
Comment (toncabral88) [inline components/ChargeBar/ChargeBar.jsx:48]: Man, here we have some typography properties. How about include width, font-size and color here too?

```suggestion
const Wrapper = styled.div`
  display: flex;
  flex-direction: column;
  font-stretch: normal;
  font-style: normal;
  font-weight: normal;
  letter-spacing: 0.18px;
  line-height: 1.5;
  
  ${({ width, background, baseFontSize }) => `
    width: ${width};
    color: ${background};
    font-size: ${baseFontSize * 0.875}px;
  `}
`;
```

File: components/ChargeBar/ChargeBar.jsx [hunk 3]
```diff
@@ -0,0 +1,151 @@
...
+        background={background}
+        width={width}
+        spacing={spacing}
+        role="progressbar"
+      >
+        <Content background={background} progress={progressPercent} />
+      </Bar>
+    </Wrapper>
+  );
+};
```
Comment (toncabral88) [inline components/ChargeBar/ChargeBar.jsx:121]: Man, I guess there are many complexity and too many elements in this simple component. I think width can be controlled by wrapper, and text elements can be wrapped by only one element. I also think span elements don't need to be styled component.

```suggestion
const InfoBlock = styled.div`
  display: flex;
  justify-content: space-between;
  
  ${({ spacing: { xsmall } }) => css`
    > * + * {
      margin-left: ${xsmall}px;
    }
  `}

  svg + span {
    flex: 1;
  }
`;

const ChargeBar = props => {
  const {
    width,
    progressPercent,
    label,
    skin,
    theme: {
      components: {
        chargeBar: {
          skins: {
            [skin]: { background },
          },
        },
      },
      spacing,
      baseFontSize,
    },
  } = props;

  return (
    <Wrapper width={width} background={background} baseFontSize={baseFontSize}>
      <InfoBlock spacing={spacing}>
        <Icon name="whatshot" size="small" skin={background} />
        <span>{label}</span>
        <span>{progressPercent}%</span>
      </InfoBlock>
      <Bar
        background={background}
        progress={progressPercent}
        spacing={spacing}
        role="progressbar"
      />
    </Wrapper>
  );
};
```

File: components/ChargeBar/ChargeBar.jsx [hunk 4]
```diff
@@ -0,0 +1,151 @@
...
+  skin: PropTypes.oneOf(['neutral', 'primary', 'secondary', 'success']),
+  theme: PropTypes.shape({
+    components: PropTypes.shape({
+      chargeBar: PropTypes.object,
+    }),
+    spacing: PropTypes.object,
+    baseFontSize: PropTypes.number,
+  }),
+  width: PropTypes.string,
+  progressPercent: PropTypes.number,
```
Comment (toncabral88) [inline components/ChargeBar/ChargeBar.jsx:147]: I didn't found a way to do this, but what do you think about include kind of range here between 0 and 100?

File: components/ChargeBar/ChargeBar.jsx [hunk 5]
```diff
@@ -0,0 +1,151 @@
...
+  progressPercent: 0,
+  skin: 'primary',
+  theme: {
+    components: {
+      chargeBar: components.chargeBar,
+    },
+    spacing: themeSpacing,
+    baseFontSize: defaultBaseFontSize,
+  },
+  label: 'Força do meu currículo',
```
Comment (toncabral88) [inline components/ChargeBar/ChargeBar.jsx:134]: By the fact this is a Quantum's component would have this default label?

**Review by toncabral88 — APPROVED** [review]
Man, I remember that we did discuss about label overlaps its area, in this case does not it have ellipsis?

## Review guide
- [x] Coverage (coverage status should not regress)\
      - `yarn test:coverage`
- [x] Unit tests (yarn test:components)
- [x] Regression \
      - first start the storybook for regression tests(and keep it open): ` yarn test:regression:storybook`; \
      - then run the regression tests: `yarn test:regression`
- [ ] Code review
- [x] TypeScript updated
- [x] a11y

### Browsers review
- [x] Layout review
  - [x] Edge
  - [x] Firefox
  - [x] Chrome
  - [x] Safari
  - [x] Mobile

File: components/ChargeBar/ChargeBar.jsx (hunk 1 above)
Comment (rapahaeru) [reply components/ChargeBar/ChargeBar.jsx:146]: i left in string to implementer choose if he passes % or px. 
What do you prefer?

File: components/ChargeBar/ChargeBar.jsx (hunk 3 above)
Comment (rapahaeru) [reply components/ChargeBar/ChargeBar.jsx:121]: Good tip! Ia agree that have too many elements. I did remove it.
But i prefere to keep some elements in styled, in intention of make code clearer.

See what i did and say what did you think

File: components/ChargeBar/ChargeBar.jsx [hunk 6]
```diff
@@ -0,0 +1,151 @@
...
+import Icon from '../Icon';
+
+const Bar = styled.div`
+  border-radius: 4px;
+  overflow: hidden;
+  position: relative;
+
+  ${({ width, background, spacing: { xxsmall } }) => `
+    background-color: ${hexToRgba(background, 0.39)};
+    height: 4px;
```
Comment (matheusrpp) [inline components/ChargeBar/ChargeBar.jsx:19]: Looking at figma, I notice that this guys In the desktop version it's a little bigger, it has a height of 8px. What do you think?

**Review by matheusrpp — CHANGES_REQUESTED** [review]
## Review guide
- [x] Coverage (coverage status should not regress)\
      - `yarn test:coverage`
- [x] Unit tests (yarn test:components)
- [x] Regression \
      - first start the storybook for regression tests(and keep it open): ` yarn test:regression:storybook`; \
      - then run the regression tests: `yarn test:regression`
- [ ] Code review
- [x] TypeScript updated
- [x] a11y

### Browsers review
- [ ] Layout review
  - [x] Edge
  - [x] Firefox
  - [x] Chrome
  - [x] Safari
  - [x] Mobile
- [ ] Ux review validation

**Review by matheusrpp — DISMISSED** [review]
## Review guide
- [x] Coverage (coverage status should not regress)\
      - `yarn test:coverage`
- [x] Unit tests (yarn test:components)
- [x] Regression \
      - first start the storybook for regression tests(and keep it open): ` yarn test:regression:storybook`; \
      - then run the regression tests: `yarn test:regression`
- [x] Code review
- [x] TypeScript updated
- [x] a11y

### Browsers review
- [x] Layout review
  - [x] Edge
  - [x] Firefox
  - [x] Chrome
  - [x] Safari
  - [x] Mobile

File: components/ChargeBar/ChargeBar.jsx (hunk 1 above)
Comment (toncabral88) [reply components/ChargeBar/ChargeBar.jsx:149]: I still prefer a number, but I guess this value should be a max-width instead width. I think this way is easier to keep component responsive.

**Review by toncabral88 — APPROVED** [review]

**Review by matheusrpp — APPROVED** [review]

**Review by toncabral88 — APPROVED** [review]

**Pull request merged**